test/integration/crud-entity/offline-crud.test.js

```diff
@@ -0,0 +1,179 @@
+runner.run(testFunc);
+
+function testFunc() {
+
+  const dataStoreTypes = [Kinvey.DataStoreType.Cache, Kinvey.DataStoreType.Sync];
+  const notFoundErrorName = 'NotFoundError';
+  const shouldNotBeCalledErrorMessage = 'Should not be called';
+
+  dataStoreTypes.forEach((currentDataStoreType) => {
+    describe(`${currentDataStoreType} Store CRUD Specific Tests`, () => {
+
+      let networkStore;
+      let syncStore;
+      let cacheStore;
+      let storeToTest;
+      const dataStoreType = currentDataStoreType;
+      const entity1 = common.getSingleEntity(common.randomString());
+      let createdUserIds = [];
+
+      before((done) => {
+        common.cleanUpAppData(collectionName, createdUserIds)
```

Comment:
I've seen this code one or two times before - perhaps it deserves its own function?